winter/web/throttling/throttling.py

```diff
@@ -33,23 +33,25 @@ def throttling(rate: Optional[str], scope: Optional[str] = None):
 class BaseRateThrottle:
     def __init__(self, throttling_: Throttling):
         self._throttling = throttling_
+        self._storage = get_throttling_statistic_storage()
 
     def allow_request(self, request: django.http.HttpRequest) -> bool:
         ident = _get_ident(request)
         key = _get_cache_key(self._throttling.scope, ident)
 
-        history = default_cache.get(key, [])
-        now = time.time()
+        with self._storage.lock(key, timeout_sec=2):
```

Comment:
I think it's better to avoid locks and rely on Redis atomic operations, something like:
```
pipe = redis.pipeline()
pipe.zadd(key, {str(now): now})
pipe.zremrangebyscore(key, 0, now - self._throttling.duration)
pipe.zcard(key)
pipe.expire(key, self._throttling.duration)
_, _, count, _ = pipe.execute()

return count <= limit
```